exchanges/order/orders.go

```diff
@@ -466,8 +466,20 @@ func (d *Detail) GenerateInternalOrderID() {
 	}
 }
 
-// Copy will return a copy of Detail
-func (d *Detail) Copy() Detail {
+// CopyValue will return a copy of Detail NOTE: This is Addressable.
+func (d *Detail) CopyValue() Detail {
+	return d.copy()
+}
+
+// CopyValueNewAddress will return the address of a new copy of the order Detail
+// WARNING: DO NOT DEREFERENCE USE METHOD CopyValue().
+func (d *Detail) CopyValueNewAddress() *Detail {
```

Comment:
This kinda thing having a copy and a copy with pointer is not something we normally do so its quite difficult to give it a name. NewAddress would imply that d.Copy doesn't have a new memory space. CopyWithPointer? CopyToPointer? I think CopyToPointer rn